x-pack/plugins/snapshot_restore/public/application/sections/home/restore_list/restore_table/restore_table.tsx

```diff
@@ -4,127 +4,94 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import React, { useEffect, useState } from 'react';
+import React, { useState, useMemo } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
 import { sortByOrder } from 'lodash';
 import { EuiBasicTable, EuiButtonIcon, EuiHealth } from '@elastic/eui';
 import { RIGHT_ALIGNMENT } from '@elastic/eui/lib/services';
+
 import { SnapshotRestore } from '../../../../../../common/types';
 import { UIM_RESTORE_LIST_EXPAND_INDEX } from '../../../../constants';
-import { useAppDependencies } from '../../../../index';
-import { uiMetricService } from '../../../../services/ui_metric';
+import { useServices } from '../../../../app_context';
 import { FormattedDateTime } from '../../../../components';
 import { ShardsTable } from './shards_table';
 
 interface Props {
   restores: SnapshotRestore[];
 }
 
-export const RestoreTable: React.FunctionComponent<Props> = ({ restores }) => {
-  const {
-    core: { i18n },
-  } = useAppDependencies();
-  const { FormattedMessage } = i18n;
-  const { trackUiMetric } = uiMetricService;
-
-  // Track restores to show based on sort and pagination state
-  const [currentRestores, setCurrentRestores] = useState<SnapshotRestore[]>([]);
-
-  // Sort state
-  const [sorting, setSorting] = useState<{
-    sort: {
-      field: keyof SnapshotRestore;
-      direction: 'asc' | 'desc';
-    };
-  }>({
-    sort: {
-      field: 'isComplete',
-      direction: 'asc',
-    },
-  });
+export const RestoreTable: React.FunctionComponent<Props> = React.memo(({ restores }) => {
+  const { i18n, uiMetricService } = useServices();
 
-  // Pagination state
-  const [pagination, setPagination] = useState({
-    pageIndex: 0,
-    pageSize: 20,
-    totalItemCount: restores.length,
-    pageSizeOptions: [10, 20, 50],
-  });
+  const [tableState, setTableState] = useState<{ page: any; sort: any }>({ page: {}, sort: {} });
 
   // Track expanded indices
-  const [itemIdToExpandedRowMap, setItemIdToExpandedRowMap] = useState<{
+  const [expandedIndices, setExpandedIndices] = useState<{
     [key: string]: React.ReactNode;
   }>({});
 
-  // On sorting and pagination change
```

Comment:
I re-wrote this component as there was an infinite re-render due to a state being update inside a `useEffect` **and** with the new eslint rule the state being updated had to be put in the array of dependencies.